### PR TITLE
Update batch README.md

### PR DESCRIPTION
**Pull request by akotlar**
1) More completely document pointing kubectl to a google cloud cluster.

2) at least in Google Cloud SDK 228.0.0, one of --zone or --region are required.

3) eval $(minikube docker-env) seemed out of place, but I may not have understood

### CODE REVIEW

File: batch/README.md [hunk 1]
```diff
@@ -19,18 +26,37 @@ minikube start
...
+```sh
+# Login with a Google account that is linked to a Google Cloud project that manages the cluster of interest
+gcloud auth login
 
-Set some environment variables so that docker images are placed in the
-`minikube` cluster's docker registry:
+gcloud projects list
+# Shows
+#PROJECT_ID             NAME             PROJECT_NUMBER
+#my-awesome-project     xxxx             xxxx
```
Comment (danking) [inline batch/README.md]: I'd put the description of what the shell block does as text outside the block, then in the block, I think we should stick to posix-y looking shell, i.e.:
```
$ gcloud projects list
PROJECT_ID             NAME             PROJECT_NUMBER
my-awesome-project     xxxx             xxxx
```

**Comment by danking** [comment]
Hmm. Maybe the env-setup script should also set the default region/zone to us-central1-a?

File: batch/README.md (hunk 1 above)
Comment (akotlar) [reply batch/README.md]: Sure. I was going for executable commands inside the code block, but see your point.

**Comment by akotlar** [comment]
> Hmm. Maybe the env-setup script should also set the default region/zone to us-central1-a?

Yep, that would remove the need for --zone or --region. We should probably document that part of the configuration, since its likely what prevented the get-credentials command from working for me as provided.

**Comment by akotlar** [comment]
Noteworthy as well. Although I can't reproduce it consistently, sometimes conda may not like `-f environment.yml`, but will work perfectly well with `conda env create`. While I prefer `-f environment.yml` because it is more obvious, dropping that argument may be less fragile. 

https://github.com/conda/conda/issues/3847

File: batch/README.md [hunk 2]
```diff
@@ -81,7 +108,7 @@ the k8s pod). Now you can load the conda environment for testing and run the
 tests against this deployment:
 
 ```
-conda env create -f environment.yaml
+conda env create -f environment.yml
```
Comment (danking) [inline batch/README.md]: nice catch

File: batch/README.md [hunk 3]
```diff
@@ -17,20 +24,40 @@ brew cask reinstall minikube # or equivalent on your OS
...
 
+1. Login with a Google account that is linked to a Google Cloud project that manages the cluster of interest
+```sh
+gcloud auth login
 ```
-gcloud container clusters get-credentials CLUSTER_NAME
+
+2. Set the active project 
+```sh
+gcloud projects list
```
Comment (danking) [inline batch/README.md]: Let's add a `#` or `$` to give a visual indication of which lines are command and which are output. I realize this makes copy-paste mildly more annoying. In other parts of the README, we've neglected to include output. I find the output useful for understanding here.

**Comment by danking** [comment]
> Noteworthy as well. Although I can't reproduce it consistently, sometimes conda may not like `-f environment.yml`, but will work perfectly well with `conda env create`. While I prefer `-f environment.yml` because it is more obvious, dropping that argument may be less fragile.
> 
> [conda/conda#3847](https://github.com/conda/conda/issues/3847)
> 
> Edit: This was actually a typo in the documentation. Fixed in [3135dc1](https://github.com/hail-is/hail/commit/3135dc124c37ec6987d96f5de0b7dbb634487b7d)

I find `conda`'s terminal UI supremely frustrating. That `conda env create -f does-not-exist` doesn't error with "file not found" blows my mind.

**Comment by danking** [comment]
🙏 https://github.com/conda/conda/pull/7385 🙏 
They finally fixed it for the forthcoming 4.6.x release.

File: batch/README.md (hunk 3 above)
Comment (akotlar) [reply batch/README.md]: Should be better

File: batch/README.md [hunk 4]
```diff
@@ -51,67 +121,67 @@ kubectl create clusterrolebinding \
   --serviceaccount=default:default
 ```
 
-Create a batch service:
+Create a `batch` service:
 
 ```
-kubectl create -f deployment.yaml
+kubectl create -f deployment.yaml.in
```
[the remark this answers is not in the document]
Comment (akotlar) [reply batch/README.md]: Sound good will modify

File: batch/README.md [hunk 5]
```diff
@@ -17,29 +23,93 @@ brew cask reinstall minikube # or equivalent on your OS
...
+Set the active proect
 
+```sh
+gcloud config set project <PROJECT_ID>
 ```
-eval $(minikube docker-env)
+
+<br>
+
+#####2. Update your `kubeconfig` to point `kubectl` to the desired cluster
```
[the remark this answers is not in the document]
Comment (akotlar) [reply batch/README.md]: Yeah, sorry. I have a broken spacebar (well, dust under the switch), which does things like this. Should improve a bit when that is fixed.

File: batch/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (akotlar) [reply batch/README.md]: Changed to make deploy

**Pull request merged**